code.gs

```diff
@@ -2026,10 +2043,13 @@ function getEventsOnDate (eventDate, calendarId) {
     log.add('The calendar with ID "' + calendarId + '" is not accessible: check your calendarId value!', Priority.FATAL_ERROR);
   }
 
+  eventDate = dateWithTimezone(year, month, day, 0, 0, 0, eventCalendar.timeZone);
+
   // Query the events calendar for events on the specified date.
   try {
-    startDate = Utilities.formatDate(eventDate, eventCalendar.timeZone, 'yyyy-MM-dd\'T\'HH:mm:ss\'Z\'');
-    endDate = Utilities.formatDate(new Date(eventDate.getTime() + 1 * 60 * 60 * 1000), eventCalendar.timeZone, 'yyyy-MM-dd\'T\'HH:mm:ss\'Z\'');
+    // Look for events from 00:00:00 to 01:00:00 of the specified day.
+    startDate = Utilities.formatDate(eventDate, eventCalendar.timeZone, 'yyyy-MM-dd\'T\'HH:mm:ssXXX');
+    endDate = Utilities.formatDate(eventDate.addDays(1), eventCalendar.timeZone, 'yyyy-MM-dd\'T\'HH:mm:ssXXX');
```

Comment:
You have changed the semantics here, maybe intentionally(?). As explained in the comment-line 2 lines up, the window of `startDate -> endDate` was
```
y/m/d 00:00:00 -> y/m/d 01:00:00
```
but it is now
```
y/m/d 00:00:00 -> y/m/d+1 00:00:00
```

It seems to me that if we match an entire day of our own timezone, we might often get matches "on one day" which match evening/morning of two different days in another timezone. I was about to say that sticking with matching a 1 hour window (`00:00 -> 01:00`) would be best, but then remembered there are [fractional-hour timezone offsets too](https://www.timeanddate.com/time/time-zones-interesting.html). So perhaps the best thing is to match a 1 minute window (`00:00 -> 00:01`). Would that confuse any of the other parts of the code? Would Google Calendar start showing non-matches? (I doubt it, but we'd need to be sure).

Incidentally, this is how I found both (A) the DST issue and (B) the dropped `+ days` issue when reviewing, because I ran `test()` on a date where I have two contacts with birthdays on consecutive days, both in Canada - and it took me ages to untangle what was really going on (in the case of (A) -> two different birthdays appearing on the same "day", and in the case of (B) -> `anticipateDays = [0, 1]` only ever matching the first birthday, "today".

Reply:
@GioBonvi So in the end, with the `.addDays(1)` syntax and the other changes you made under-the-hood, is the discussed issue regarding +1day versus +1hour versus +1minute (allowing for DST) now a moot-point? I lost track a bit. If so then I will go ahead and re-review everything (lint, tests, etc...). If not, I'll wait until you update this bit of code.

Reply:
+1 day was an error (corrected in 874cbcd). I changed it to +1 minute as you suggested after performing some tests: it works OK on my account.
I've also discoverd another (unrelated) problem by chance: parseInt() had troubles parsing '09' because strings with leading zeroes are interpreted as octal and 09 is not a valid octal number.